tap_mysql/sync_strategies/common.py

```diff
@@ -47,6 +49,37 @@ def monkey_patch_date(date_str):
 #--------------------------------------------------------------------------------------------
 #--------------------------------------------------------------------------------------------
 
+# boolean function to check if the error is 'timeout' error or not
+def is_timeout_error():
+    """
+        This function checks whether the URLError contains 'timed out' substring and return boolean
+        values accordingly, to decide whether to backoff or not.
+    """
+    def gen_fn(exc):
+        if str(exc).__contains__('timed out'):
+            # retry if the error string contains 'timed out'
+            return False
+        return True
+
+    return gen_fn
+
+def reconnect(details):
```

Comment:
why we need this? please add code comment

Reply:
Added comment in the code.